whenchanged/whenchanged.py

```diff
@@ -135,49 +138,64 @@ def run(self):
 def print_usage(prog):
     print(__doc__ % {'prog': prog}, end='')
 
+def parse_args(argv):
+    """
+    Parse the various args and returns a WhenChanged object.
 
-def main():
-    args = sys.argv
-    prog = os.path.basename(args.pop(0))
-
-    if '-h' in args or '--help' in args:
-        print_usage(prog)
-        exit(0)
+    The function can return None if the arguments were correct but
+    could not build an object (eg display help).
 
+    The function raises ValueError in case of bad arguments.
+    """
     files = []
     command = []
     recursive = False
     verbose = False
     run_once = False
     run_at_start = False
 
-    while args and args[0][0] == '-':
-        flag = args.pop(0)
-        if flag == '-v':
+    prog = os.path.basename(argv.pop(0))
+    args, shiftargs = gnu_getopt(argv, 'hvr1sc:', longopts=["help"])
+    for arg,argopt in args:
+        if arg == '-h' or arg == '--help':
+            print_usage(prog)
+            return None
+        elif arg == '-v':
             verbose = True
-        elif flag == '-r':
+        elif arg == '-r':
             recursive = True
-        elif flag == '-1':
+        elif arg == '-1':
             run_once = True
-        elif flag == '-s':
+        elif arg == '-s':
             run_at_start = True
-        elif flag == '-c':
-            command = args
-            args = []
+        elif arg == '-c':
+            # special case: all following arguments are part of a command
+            r_c = re.compile(r'\-[^\-]*c(.*)')
```

Comment:
It sure is because it has no way to specify that an option treats all the following arguments as option argument. The thing that getopt does well is managing options in a standard way, though. IMHO it is trickier to get option parsing right than working around getopt for that specificity. As a result I find the code clearer than before, but that may well be a question of personnal taste.